include/FilesystemManagement.h

```diff
@@ -0,0 +1,156 @@
+    /// @param FilePath The path and name of the file to check for.
+    /// @return Returns true if the file at the specified location exists, false otherwise.
+    [[nodiscard]]
+    Boole MEZZ_LIB FileExists(const StringView FilePath);
+
+    /// @brief Copies a file on disk to a new location.
+    /// @note This function makes no attempt to copy file permissions or attributes, only data.
+    /// @param OldFilePath The existing path to the file (including the filename) to be copied.
+    /// @param NewFilePath The path (including the filename) to where the file should be copied.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully copied, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CopyFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Moves a file on disk from one location to another.
+    /// @remarks This function can be used to rename files.
+    /// @param OldFilePath The existing path to the file (including the filename) to be moved.
+    /// @param NewFilePath The path (including the filename) to where the file should be named.
+    /// @param FailIfExists If true the operation will fail if a file with the target name already exists.
+    /// @return Returns true if the file was successfully moved, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB MoveFile(const StringView OldFilePath, const StringView NewFilePath,
+                                   const Boole FailIfExists);
+    /// @brief Deletes a file existing on the filesystem.
+    /// @param FilePath The existing path to the file (including the filename) to be deleted.
+    /// @return Returns true if the operation was successful, false if it failed.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB RemoveFile(const StringView FilePath);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Symlinks
+
+    /// @brief Creates a symbolic link to a file on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateSymlink(const StringView SymPath, const StringView TargetPath);
+    /// @brief Creates a symbolic link to a directory on disk.
+    /// @param SymPath A path (including the name of the symbolic link) to where the link should be placed.
+    /// @param TargetPath A path to where the symbolic link will point to.
+    /// @return Returns true if the operation was successful, false otherwise.
+    [[nodiscard]]
+    ModifyResult MEZZ_LIB CreateDirectorySymlink(const StringView SymPath, const StringView TargetPath);
```

Comment:
Why is there no function to check if a file is a symlink?

Reply:
It exists now.